python/lsst/afw/detection/footprintSet.cc

```diff
@@ -102,6 +102,18 @@ PYBIND11_PLUGIN(_footprintSet) {
     clsFootprintSet.def("swap", &FootprintSet::swap);
     //    clsFootprintSet.def("swapFootprintList", &FootprintSet::swapFootprintList);
     clsFootprintSet.def("setFootprints", &FootprintSet::setFootprints);
+    // The pybind11 wrapped getFootprints dereferences the shared pointer to the std::vector of Footprints
+    // held by the FootprintSet when returning to python to work around a stl caster issue. However this
+    // creates a problem for the setter that now expects a shared pointer, making the setter unable to
+    // operate on the output form the getter. Below defines two overrides for the setter function, one
+    // that can handle a shared pointer (in case somewhere else in the stack has one somehow) and one
+    // that will accept a standard vector of Footprints, which will be internally placed in a shared
+    // pointer to match the underlying C++ function signature
+    clsFootprintSet.def("setFootprints",
+        [](FootprintSet &self, std::shared_ptr<FootprintSet::FootprintList> footListPtr)
+        {self.setFootprints(footListPtr);});
```

Comment:
> ... there is no difference to the standard `&FootprintSet::setFootprints`. Where the difference lies is that if you have two function signatures, you must do a function pointer cast to disambiguate

Does this mean we can just delete the `clsFootprintSet.def("setFootprints", &FootprintSet::setFootprints);` line entirely? If I'm understanding the above, it's now unused.

To the second point, I'd tend to only include code that we explicitly are using and (vitally!) testing. However, I don't feel strongly enough about this to argue that you should take it out.